script.js

```diff
@@ -1,12 +1,216 @@
 //You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+let titleHeader = document.querySelector("h4")
+let imageHolder = document.querySelector(".photo")
+let subtitleHolder = document.querySelector('.episode-code');
+let descriptionHolder = document.querySelector('.item-text');
+let sectionCenter = document.querySelector('.section-center');
+let searchBar = document.querySelector("#searchBar");
+let selectEpisodes = document.querySelector("#select-episode");
+let goBackToAllEpisodes = document.querySelector(".go-back-to-all-episodes");
+let selectShows = document.querySelector("#select-shows")
+
+let episodesData = [];
+let showsData = [];
+
+
+//1 populate selectShows input with shows ..
+//all episodes must be sorted in alphabetical order
+
+
+
+
+
+
+function populateShows() {
+  selectShows.options.length = 0;
+  
+
+  getAllShows().sort((a, b) => (a.name > b.name) ? 1 : ((b.name > a.name) ? -1 : 0)).forEach((item) => {
+
+    let opt = item.name;;
+   
+    let el = document.createElement("option");
+    el.textContent = opt;
+    el.value = opt;
+    selectShows.appendChild(el);
+
+  });
 }
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+
+//sort the shows
+
+
+
+///Now I have all the shows so need to  get access to  one show which user selected in select box.
+
+function selectShowsToGetEpisodes() {
+  document.querySelector("#select-shows").addEventListener('change', (e) => {
+    const selectString = e.target.value;
+    console.log(selectString);
+    let selectedShow = getAllShows().filter((item) => {
+      
+      return (
+        item.name === selectString);
+    });
+    
+    let idOfShow = selectedShow.map(data => data.id);
+  
+    //Now I have an access to a particular shows that user serlect from the select box so I need 
+    //to go and fetch episodes of that selected show.
+
+    //fetch an api based on selcted show
+    fetch(`https://api.tvmaze.com/shows/${idOfShow}/episodes`)
```

Comment:
Well done using template literal to insert the `id` into this string. So piece of code is important as it is doing the fetch for all the episodes - I'd consider wrapping up this `fetch` in a function called something like `fetchEpisodes()` and then make it take an `id`

Reply:
It will be easier for you to reason about and then you can also reuse it in other parts of your code if you need to